### DISCUSSION TITLE
Tools and CI Meeting - Feb 12, 2024

### DISCUSSION THREAD

**Discussion by spbrogan** [tianocore/edk2#5158] [General]
Weekly meeting to discuss Tools, CI, and codebase construction topics.

## Status

Pending

## Agenda

None - Submit topics by replying to this discussion to confirm meeting.

## Groups.io meeting invite: 

https://edk2.groups.io/g/devel/viewevent?repeatid=43269&eventid=2159616&calstart=2024-01-08

## Microsoft Teams meeting

Join on your computer or mobile app
[Click here to join the meeting](https://teams.microsoft.com/l/meetup-join/19%3ameeting_ZDI2ZDg4NmMtMjI1My00MzI5LWFmYjAtMGQyNjUzNTBjZGYw%40thread.v2/0?context=%7b%22Tid%22%3a%2272f988bf-86f1-41af-91ab-2d7cd011db47%22%2c%22Oid%22%3a%2223af6561-6e1c-450d-b917-d9d674eb3cb6%22%7d)
Or call in (audio only)
+1 425-616-0754,,330776518# United States, Seattle
Phone Conference ID: 330 776 518#
Find a local number | [Reset PIN](https://mysettings.lync.com/pstnconferencing)
[Learn More](https://aka.ms/JoinTeamsMeeting) | [Meeting options](https://teams.microsoft.com/meetingOptions/?organizerId=23af6561-6e1c-450d-b917-d9d674eb3cb6&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&threadId=19_meeting_ZDI2ZDg4NmMtMjI1My00MzI5LWFmYjAtMGQyNjUzNTBjZGYw@thread.v2&messageId=0&language=en-US)

**Comment by spbrogan** [comment]
No meeting Feb 5th as still no topics requested.

**Comment by spbrogan** [comment]
Agenda for Feb 12th. 

Agenda topics: 
1. CI and Package DSC files.  
    * https://edk2.groups.io/g/devel/message/115244
    * https://edk2.groups.io/g/devel/message/115185
2. Basetools Project status 
    * https://github.com/tianocore/edk2-basetools
3. Commits spanning Packages.
   * https://edk2.groups.io/g/devel/message/114915

**Comment by spbrogan** [comment]
Meeting video shared here: https://youtu.be/jNlfxZr7GzQ

Generated by AI. 

Meeting notes:

-   CI build process and package DSC files:Sean and Kinney disagreed on
    whether to delete the non-continuous integration path of the DSC
    files or to add a check for it in CI. No conclusion reached.

-   Base tools project status and issues:Kinney agreed to talk to
    Rebecca, Liming and Christine about their commitment to maintain and
    update the base tools project, which is currently broken and behind
    the EDK 2 code. Sean questioned the value and feasibility of the
    project.

-   Commits spanning packages:Michael proposed to add a CI check to
    prevent commits that span multiple packages, which can cause
    bisectability and downstream issues. Kinney agreed and asked Michael
    to write a BZ for it.

-   Unit tests and mock libraries:Aaron raised some questions and
    challenges about using mock libraries and PCDs for unit testing.
    Kinney pointed to some examples and documents for guidance and
    mentioned a script to auto-generate mocks. Sean suggested to improve
    the documentation and alignment of the unit test framework.

Follow-up tasks:

-   CI build process and package DSC files:Discuss the issue of
    continuous integration flag and package DSC files on the mailing
    list and try to reach a consensus (Sean, Kinney, Joey, and other
    interested parties)

-   Base tools project:Talk to Rebecca, Liming, and Christine about
    their ability to maintain and update the base tools project and fix
    the CI issues (Kinney)

-   Commits spanning packages:Write up a bugzilla issue with the
    suggested algorithm to check for commits spanning packages and
    assign it to potential resources (Michael)

-   Unit test framework:Review the unit test framework document and
    provide feedback on the Python coding style and best practices
    (Joey)

-   Unit test framework:Share the script to auto-generate mocks from
    library include files and consider making it part of an automation
    step (Kinney)

-   Stack cookie functionality:Look at the PR and the build spec for the
    issue of no class libraries and stack cookie functionality and
    provide feedback (Kinney)

-   Meeting agenda and notification:Send out the call for topics or
    queued up topics a day or two ahead of the meeting and update the
    calendar if the meeting is cancelled (Sean)

**Comment by ghost** [comment]
I'd like to hear where we stand with the project to move BaseTools applications from C to Python.

**Reply by spbrogan** [reply]
Great question.  I'll move it to the vNext discussion topic.  I would expect the appropriate way for the project to express this info is by updating the related issues listed here.  https://github.com/tianocore/edk2-basetools/issues 

In general, the edk2-basetools project is not leveraging the tools GitHub provides but then that is par for the course with tianocore.
